driver/compiler_hooks.mli

```diff
@@ -0,0 +1,48 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 1996 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+open Misc
+open Compile_common
+
+type _ language =
+  | Parse_tree_intf : Parsetree.signature language
+  | Parse_tree_impl : Parsetree.structure language
+  | Typecheck_intf : Typedtree.signature language
+  | Typecheck_impl : (Typedtree.structure * Typedtree.module_coercion) language
+  | Raw_lambda : Lambda.program language
+  | Lambda : Lambda.program language
+  (* CR-someday poechsel: use flambda2 *)
+  | Raw_flambda2 : Flambda2.Flambda_unit.t language
+  | Flambda2 : Flambda2.Flambda_unit.t language
+  | Raw_flambda1 : Flambda.program language
+  | Flambda1 : Flambda.program language
+  | Raw_clambda : Clambda.ulambda language
+  | Clambda : Clambda.ulambda language
+
+  | Mach_combine : Mach.fundecl language
```

Comment:
This is a (language * pass) and the intent is the result of a particular pass, which is not necessarily the same as the input to that pass.  I don't know if "pass" would be a better name than "language" here, perhaps just add some comments?
There is some code in Clflags.Compiler_pass and Compiler_ir that should use the same names.

Reply:
I quite like `Pass`